src/scikit_package/cli/gh.py

```diff
@@ -14,6 +19,32 @@ def _get_issue_content(issue_url):
     issue_content: dict
         issue-title and issue-body to be broadcast.
     """
-    source_repo_url = None
-    issue_content = None
+    parsed = urlparse(issue_url)
+    path_parts = parsed.path.strip("/").split("/")
+    try:
+        owner = path_parts[0]
+        repo = path_parts[1]
+        issue_number = int(path_parts[3])
+    except (IndexError, ValueError):
+        raise ValueError(
+            f"{issue_url} is not a valid url to be parsed. "
+            "Please input the url of the issue to be broadcasted. "
+            "Its format should be https://"
+            "github.com/username/reponame/issues/issue-number"
+        )
+    api_url = (
```

Comment:
it is not clear to me why you are doing this.  We are taking in a url, you are then decomposing it and then recomposing it.  Why don't we simply pass the url directly to the request?

I see that you are handing back the source directory and we need to do some composing for that to happen, but why are we doing that?   Let's try and delete all that work.